src/arche/tools/s3.py

```diff
@@ -42,5 +42,7 @@ def get_contents_as_string(bucket: str, filepath: str) -> str:
 
 
 def get_contents(url: str) -> str:
+    if urlparse(url).scheme == "file":
```

Comment:
I wonder, however, if instead of forbidding ‘file’ we should forbid anything but ‘http’ and ‘https’.

Reply:
Indeed. I am not aware of anybody using `http` so I left it out. Also I moved the parsing logic to s3.py, I think it belongs there.